birdhouse/config/jupyterhub/jupyterhub_config.py.template

```diff
@@ -187,4 +187,37 @@ blocked_users = {'authtest', '${CATALOG_USERNAME}', 'anonymous'}
 c.Authenticator.blacklist = blocked_users  # v0.9+
 c.Authenticator.blocked_users = blocked_users  # v1.2+
 
+
+# ------------------------------------------------------------------------------
+# Shutdown idle user server based on configured timeouts.
+# ------------------------------------------------------------------------------
+# Timeout (in seconds) to shut down the whole server when no kernels or terminals are running and there is no activity.
+# If left undefined or zero, the feature will not be enabled.
+jupyter_idle_server_cull_timeout = int("${JUPYTER_IDLE_SERVER_CULL_TIMEOUT}" or 0)
+if jupyter_idle_server_cull_timeout:
+    c.Spawner.args.append('--NotebookApp.shutdown_no_activity_timeout={}'.format(jupyter_idle_server_cull_timeout))
+# Timeout (in seconds) after which individual user kernels/terminals are considered idle and ready to be culled.
+jupyter_idle_kernel_cull_timeout = int("${JUPYTER_IDLE_KERNEL_CULL_TIMEOUT}" or 0)
+# Interval (in seconds) on which to check for idle kernels exceeding the cull timeout value.
+jupyter_idle_kernel_cull_interval = int("${JUPYTER_IDLE_KERNEL_CULL_INTERVAL}" or 0)
+if jupyter_idle_kernel_cull_timeout:
+    if not jupyter_idle_kernel_cull_interval or jupyter_idle_kernel_cull_interval > jupyter_idle_kernel_cull_timeout:
+        jupyter_idle_kernel_cull_interval = jupyter_idle_kernel_cull_timeout / 2
+    c.Spawner.args.extend([
+        '--MappingKernelManager.cull_idle_timeout={}'.format(jupyter_idle_kernel_cull_timeout),
+        '--MappingKernelManager.cull_interval={}'.format(jupyter_idle_kernel_cull_interval),
```

Comment:
Funny I didn't have to manually set the `cull_interval` to haft of the `cull_idle_timeout` and it was working fine.  What is the default value of the 2 `cull_interval` if not set?

Reply:
[300 seconds](https://jupyter-server.readthedocs.io/en/latest/api/jupyter_server.services.kernels.html#jupyter_server.services.kernels.kernelmanager.MappingKernelManager.cull_interval_default) I believe

Reply:
Indeed, there is the default of 300 from jupyter-server, but with my proposed changes at L205, the value is automatically halved from the timeout if not specified. I had to do this, otherwise the interval between updates took at least 300s regardless of my timeout=10 test.